modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -1328,4 +1342,325 @@ public int deleteAllMessageMetadataFromDLC(String storageQueueName, String DLCQu
+            throw new AndesException("Error occurred while adding retained message content to DB ", e);
+        } finally {
+            close(updateMetadataPreparedStatement, RDBMSConstants.TASK_STORING_RETAINED_MESSAGE_PARTS);
+            close(deleteContentPreparedStatement, RDBMSConstants.TASK_STORING_RETAINED_MESSAGE_PARTS);
+            close(insertContentPreparedStatement, RDBMSConstants.TASK_STORING_RETAINED_MESSAGE_PARTS);
+            close(connection, RDBMSConstants.TASK_STORING_RETAINED_MESSAGE_PARTS);
+        }
+    }
+
+
+    /**
+     * Used to store details about a retained item entry
+     */
+    private static class RetainedItemData {
+        /**
+         * Topic id of the DB entry
+         */
+        public int topicID;
+
+        /**
+         * Retained message ID for the topic
+         */
+        public long messageID;
+
+        private RetainedItemData(Integer topicID, long messageID) {
+            this.topicID = topicID;
+            this.messageID = messageID;
+        }
+    }
+
+    /**
+     * Update batching prepared statements with current message
+     *
+     * @param updateMetadataPreparedStatement
+     *         update prepared statement
+     * @param deleteContentPreparedStatement
+     *         delete prepared statement
+     * @param insertContentPreparedStatement
+     *         insert prepared statement
+     * @param message
+     *         current message
+     * @param metadata
+     *         current message metadata
+     * @param retainedItemData
+     *         retained item data
+     * @throws SQLException
+     */
+    private void addRetainedMessageToUpdateBatch(PreparedStatement updateMetadataPreparedStatement,
+                                                 PreparedStatement deleteContentPreparedStatement,
+                                                 PreparedStatement insertContentPreparedStatement,
+                                                 AndesMessage message,
+                                                 AndesMessageMetadata metadata, RetainedItemData retainedItemData)
+            throws SQLException {
+        // update metadata
+        updateMetadataPreparedStatement.setLong(1, metadata.getMessageID());
+        updateMetadataPreparedStatement.setBytes(2, metadata.getMetadata());
+        updateMetadataPreparedStatement.setInt(3, retainedItemData.topicID);
+        updateMetadataPreparedStatement.addBatch();
+
+        // update content
+        deleteContentPreparedStatement.setLong(1, retainedItemData.messageID);
+        deleteContentPreparedStatement.addBatch();
+        for (AndesMessagePart messagePart : message.getContentChunkList()) {
+            insertContentPreparedStatement.setLong(1, metadata.getMessageID());
+            insertContentPreparedStatement.setInt(2, messagePart.getOffSet());
+            insertContentPreparedStatement.setBytes(3, messagePart.getData());
+            insertContentPreparedStatement.addBatch();
+        }
```

Comment:
These are not magic numbers. This is default way of supplying values for preparedStatement parameters. Please refer storeMessagePart, getContent methods in RDBMSMessageStoreImpl. To avoid these we might have to revamp RDBMSMessageStoreImpl class as a separate task.